sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/queries/sql/NexmarkSqlQuery.java

```diff
@@ -46,9 +44,8 @@ public NexmarkSqlQuery(NexmarkConfiguration configuration,
   protected PCollection<KnownSize> applyPrim(PCollection<Event> events) {
     PCollection<Row> queryResults = events.apply(queryTransform);
 
-    PCollection<? extends KnownSize> resultRecordsSizes = queryResults
-        .apply(RowSize.parDo())
-        .setCoder(RowSize.CODER);
+    PCollection<? extends KnownSize> resultRecordsSizes =
+        queryTransform.applyModel(queryResults);
```

Comment:
There's only about 3 pojo models we're interested in, so if you make `RowSize.parDo()` convert to those models  instead of converting to a custom `RowSize` (see my other comment), then you don't need to change the interfaces or inheritance